### ISSUE TITLE
Strange merging behavior with insert into newline

### ISSUE THREAD

**Issue by ffd8** [ace-diff/ace-diff#93]
Just ran into a strange issue where merging makes some mistakes on `{}` brackets.
Using your [demo1](https://ace-diff.github.io/ace-diff/demos/demo1/), tested the following:

#### Left editor:
```js
let flipSpeed = 55

function setup() {
	createCanvas(windowWidth, windowHeight)
	background(255)
}

function draw() {
	background(0)
}
```

#### Right editor:
```js
function setup() {
	createCanvas(windowWidth, windowHeight)

}

function draw() {

}
```

#### What it looks like initially:
![image](https://user-images.githubusercontent.com/570957/148664988-0833fadd-d532-4b23-8eba-c7db33fa3e00.png)

#### After the first two merges (shows mistake):
![image](https://user-images.githubusercontent.com/570957/148665007-f25ab5c6-550a-4037-8156-fe55776d20b0.png)

#### Then corrects itself if I add an extra line (only if removing the indented tab):
![image](https://user-images.githubusercontent.com/570957/148665026-2bbcb507-5bcb-4b83-9d7d-736a3c4f9047.png)

**Comment by neeraj-chhatani** [comment]
How to use Ace diff in angular

**Comment by jmthomas** [comment]
I don't think this is due to the braces. It's due to the newlines. It appears like it is inserting the line instead of replacing the blank line. The entire ace-diff library needs to handle newlines better.
